src/index.js

```diff
@@ -89,10 +92,10 @@ class DelegatedContentRouting {
   async provide (key) {
     const keyString = key.toBaseEncodedString()
     log('provide starts: ' + keyString)
-    await this._httpQueueRefs.add(() =>
-      this.refs(keyString, { recursive: false })
+    const results = await this._httpQueueRefs.add(() =>
+      all(this.refs(keyString, { recursive: false }))
```

Comment:
Not seeing the added return, should be on the `all`
```suggestion
      return all(this.refs(keyString, { recursive: false }))
```

Reply:
that return is implicit, as we are not using `{` `}`. I added a return form the queue operation

Reply:
ah yeah, you're right, I was looking at the newline and not the brackets. This is gtg.